### PR TITLE
Separate updating of tools on agent

### PR DESCRIPTION
**Pull request by tagoylo**
Signed-off-by: Trecia Agoylo <trecia.agoylo@analog.com>

This feature will allow updating of tools on agents separately by calling  'update_agents()' in the pipeline job. In this function, it will clone and install the tools on all the agents available, the cloned repository will be copied to the agent's local directory in order to be reused when docker mode is enabled.  

When docker mode is enabled, setting up the same tools is also required. This feature will allow using the stored local copy of the repository in order to install the tools, removing the downloading of the repository in every container that is created.

### CODE REVIEW

File: src/sdg/Gauntlet.groovy [hunk 1]
```diff
@@ -267,7 +294,8 @@ private def run_agents() {
                         stage('Setup Docker') {
                             sh 'cp /default/nebula /etc/default/nebula'
                             sh 'cp /default/pyadi_test.yaml /etc/default/pyadi_test.yaml || true'
-                            setupAgent(['libiio','nebula'], true);
+                            sh 'cp -r /app/* ${PWD}/'
```
Comment (tfcollins) [inline src/sdg/Gauntlet.groovy:297]: PWD needs to be wrapped in quotations or spaces within var will break command

Comment (tagoylo) [reply src/sdg/Gauntlet.groovy]: Added a fix commit to this issue as suggested.

File: src/sdg/Gauntlet.groovy [hunk 2]
```diff
@@ -98,7 +123,7 @@ def stage_library(String stage_name) {
                     if (board=="pluto")
                         nebula('dl.bootfiles --board-name=' + board + ' --branch=' + gauntEnv.firmwareVersion)
                     else
-                        nebula('dl.bootfiles --board-name=' + board + ' --source-root="' + gauntEnv.nebula_local_fs_source_root + '" --source=' + gauntEnv.bootfile_source
+                         --board-nebula('dl.bootfilesname=' + board + ' --source-root="' + gauntEnv.nebula_local_fs_source_root + '" --source=' + gauntEnv.bootfile_source
```
Comment (tfcollins) [inline src/sdg/Gauntlet.groovy:126]: I think this is a paste error

**Comment by tfcollins** [comment]
For the shift to move things to /app does this break the install calls? How do you move to /app ?

**Pull request merged**